### PR TITLE
Add `suffix` option to `embed_templates`

### PR DESCRIPTION
**Pull request by jsonmaur**
I've been using the new `embed_templates` macro to compile Swoosh emails with `Phoenix.Template.render`. It has been super useful for using HEEx in email templates. A stripped down example:

```elixir
defmodule MyAppWeb.AccountsNotifier do
  import Swoosh.Email

  embed_templates("accounts_notifier/*")

  def welcome(assigns) do
    html = Phoenix.Template.render(__MODULE__, "welcome", "html", Map.to_list(assigns))

    new()
    |> to(assigns.email)
    |> subject("Welcome to My App")
    |> html_body(html)
  end
end
```

Since the `embed_templates` macro generates `welcome/1` as well, I have to rename my `welcome/1` to something like `welcome_email/1` to prevent a name conflict. My proposed solution is to simply add a `:keep_format` option to `embed_templates`. This would make the macro more flexible in other situations, and allow for something like this:

```elixir
defmodule MyAppWeb.AccountsNotifier do
  import Swoosh.Email

  embed_templates("accounts_notifier/*", keep_format: true)

  def welcome(assigns) do
    html = Phoenix.Template.render(__MODULE__, "welcome.html", "html", Map.to_list(assigns))

    new()
    |> to(assigns.email)
    |> subject("Welcome to My App")
    |> html_body(html)
  end
end
```

### CODE REVIEW

**Comment by josevalim** [comment]
I would prefer a :suffix option. This way you can do: `suffix: ”_template”` or similar. WDYT?

**Comment by jsonmaur** [comment]
That definitely makes it more flexible. The main use case I had in mind was being able to embed different formats without naming conflicts, like so:

```elixir
defmodule MyAppWeb.AccountsNotifier do
  import Swoosh.Email

  embed_templates("accounts_notifier/*")

  def welcome(assigns) do
    assigns = Map.to_list(assigns)

    html = Phoenix.Template.render(__MODULE__, "welcome_html", "html", assigns)
    text = Phoenix.Template.render(__MODULE__, "welcome_text", "text", assigns)

    new()
    |> to(assigns.email)
    |> subject("Welcome to My App")
    |> html_body(html)
    |> text_body(text)
  end
end
```

Although looking at it again, I think https://github.com/phoenixframework/phoenix_live_view/commit/7e7e8eee854659e45d4c367d7ea902c19f40e88d might make this obsolete with the hardcoded `.html` suffix on the pattern. What would you think about making that suffix configurable, and attaching the suffix to the function names only when it's defined? For example, if I had `accounts_notifier/welcome.html.heex` and `accounts_notifier/welcome.text.eex`:

```elixir
defmodule MyAppWeb.AccountsNotifier do
  embed_templates("accounts_notifier/*")

  # welcome(assigns)
end

defmodule MyAppWeb.AccountsNotifier do
  embed_templates("accounts_notifier/*", suffix: [:html, :text])

  # welcome_html(assigns)
  # welcome_text(assigns)
end
```

**Comment by josevalim** [comment]
I would like the suffix to be any string, regardless of the template name and format. This way you can even call embed_templates multiple times for different formats with different suffixes.

**Comment by jsonmaur** [comment]
Makes sense. What do you think about dropping the hardcoded `.html` extension alongside adding the suffix option to allow for calling `embed_templates` multiple times for different formats? See the latest commit.

File: lib/phoenix_component.ex [hunk 1]
```diff
@@ -1401,14 +1404,22 @@ defmodule Phoenix.Component do
...
+      end
   """
   defmacro embed_templates(pattern, opts \\ []) do
     quote do
       Phoenix.Template.compile_all(
-        &Phoenix.Component.__embed__/1,
+        &Phoenix.Component.__embed__(&1, unquote(opts)[:suffix] || ""),
         Path.expand(unquote(opts)[:root] || __DIR__, __DIR__),
-        unquote(pattern) <> ".html"
+        unquote(pattern)
```
Comment (josevalim) [inline lib/phoenix_component.ex:1422]: This should not be changed. :)

```suggestion
        unquote(pattern) <> ".html"
```

File: lib/phoenix_component.ex [hunk 2]
```diff
@@ -1439,7 +1450,13 @@ defmodule Phoenix.Component do
...
 
   @doc false
-  def __embed__(path), do: path |> Path.basename() |> Path.rootname() |> Path.rootname()
+  def __embed__(path, suffix),
+    do:
+      path
+      |> Path.basename()
+      |> Path.rootname()
+      |> Path.rootname()
+      |> String.replace_suffix("", suffix)
```
Comment (josevalim) [inline lib/phoenix_component.ex:1459]:
```suggestion
      |> Kernel.<>(suffix)
```

File: lib/phoenix_component.ex (hunk 1 above)
Comment (josevalim) [reply lib/phoenix_component.ex:1422]: Oh, you want to explicitly embed `.text` ones. So maybe we also need a `:format` option? /cc @chrismccord

**Comment by leandrocp** [comment]
FWIW adding `:format` option would solve https://github.com/phoenixframework/phoenix_live_view/pull/2360 too :)

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**